### ISSUE TITLE
Regression in UX, probably related to piSort

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#3982]
Given this WIP code, shrank from a larger development:
```agda
module _ {a} {A : Set a} where

open import Agda.Builtin.Equality
open import Agda.Builtin.List

infix 4 _⊆_

postulate
  _⊆_ : (xs ys : List A) → Set a
  ⊆-trans : ∀{xs ys zs} → xs ⊆ ys → ys ⊆ zs → xs ⊆ zs

private
  variable
    xs ys zs : List A
    σ τ : ys ⊆ zs
    x y : A
    x≈y : x ≡ y

lemma : ∀ us (ρ : us ⊆ zs) (τ' : x ∷ xs ⊆ us) (σ' : ys ⊆ us)
         (τ'∘ρ≡τ : ⊆-trans τ' ρ ≡ τ)
         (σ'∘ρ≡σ : ⊆-trans σ' ρ ≡ σ)
      → us ⊆ us
lemma {τ = τ} {σ = σ} us ρ τ' σ' τ'∘ρ≡τ σ'∘ρ≡σ =
  lem {τ = τ} {σ} us ρ τ' σ' τ'∘ρ≡τ σ'∘ρ≡σ
  where
    lem : ∀ {ys zs} {τ : (x ∷ xs) ⊆ zs} {σ : ys ⊆ zs}
         us (ρ : us ⊆ zs) (τ' : x ∷ xs ⊆ us) (σ' : ys ⊆ us)
         (τ'∘ρ≡τ : ⊆-trans τ' ρ ≡ τ)
         (σ'∘ρ≡σ : ⊆-trans σ' ρ ≡ σ) →
       {!!}
    lem = {!!}
```
With 2.6.0, I get an unsolved constraint related to the unspecified target type of `lem` which is not solved by unification.  I can still work on the file, like use `C-c C-c` to introduce arguments to `lem` and to split them etc.
This is what 2.6.0 shows me, which is quite comprehensible:
```
?0 : Set a
?1
  : (us₁ : List A) (ρ₁ : us₁ ⊆ zs) (τ'' : x ∷ xs ⊆ us₁)
    (σ'' : ys ⊆ us₁) (τ'∘ρ≡τ₁ : ⊆-trans τ'' ρ₁ ≡ τ₁)
    (σ'∘ρ≡σ₁ : ⊆-trans σ'' ρ₁ ≡ σ₁) →
    _107 (us = us) (us = us₁)
_107 : Set a  [ at 32,8-12 ]
_108 : us ⊆ us  [ at 26,3-43 ]
_109 : us ⊆ us  [ at 26,3-43 ]

———— Errors 
Failed to solve the following constraints:
  _108
    := λ {a} {A} {zs} {x} {xs} {ys} {τ} {σ} us ρ τ' σ' τ'∘ρ≡τ σ'∘ρ≡σ →
         lem us ρ τ' σ' τ'∘ρ≡τ σ'∘ρ≡σ
    [blocked on problem 156]
  [156] _107 (us = us) =< us ⊆ us : Set a
```

With current master I get a hard and not very comprehensible error:
```
Cannot instantiate the metavariable _70 to solution
piSort (Set a)
(λ x →
   piSort (Set a)
   (λ xs →
      piSort (Set a)
      (λ ys →
         piSort (Set a)
         (λ zs →
            piSort (Set a)
            (λ τ₁ →
               piSort (Set a)
               (λ σ₁ →
                  piSort (Set a)
                  (λ us₁ →
                     piSort (Set a)
                     (λ ρ₁ →
                        piSort (Set a)
                        (λ τ'' →
                           piSort (Set a)
                           (λ σ'' →
                              piSort (Set a)
                              (λ τ'∘ρ≡τ₁ →
                                 piSort (Set a)
                                 (λ σ'∘ρ≡σ₁ →
                                    _99 (us = us) (ρ = ρ) (τ' = τ') (σ' = σ') (τ'∘ρ≡τ = τ'∘ρ≡τ)
                                    (σ'∘ρ≡σ = σ'∘ρ≡σ) (us = us₁) (ρ = ρ₁) (τ' = τ'') (σ' = σ'')
                                    (τ'∘ρ≡τ = τ'∘ρ≡τ₁) (σ'∘ρ≡σ = σ'∘ρ≡σ₁)))))))))))))
since it contains the variable a
which is not in scope of the metavariable or irrelevant in the metavariable but relevant in the solution
when checking that the expression
∀ {ys} {zs} {τ = τ₁ : x ∷ xs ⊆ zs} {σ = σ₁ : ys ⊆ zs} us₁
  (ρ₁ : us₁ ⊆ zs) (τ'' : x ∷ xs ⊆ us₁) (σ'' : ys ⊆ us₁)
  (τ'∘ρ≡τ₁ : ⊆-trans τ'' ρ₁ ≡ τ₁) (σ'∘ρ≡σ₁ : ⊆-trans σ'' ρ₁ ≡ σ₁) →
?
has type _70
```
